src/lib/codemirror-editor.ts

```diff
@@ -134,16 +214,47 @@ export class CodeMirrorEditorElement extends LitElement {
         this.shadowRoot!.appendChild(dom);
       },
       {
-        value: this.value,
+        value: this.value ?? '',
         lineNumbers: this.lineNumbers,
         mode: this._getLanguageMode(),
+        theme: this.theme,
+        readOnly: this.readonly,
       }
     );
     cm.on('change', () => {
       this._value = cm.getValue();
       this.dispatchEvent(new Event('change'));
     });
     this._codemirror = cm;
+    this._setBackgroundColor();
+  }
+
+  /**
+   * We want the CodeMirror theme's background color to win if
+   * "--playground-editor-background-color" is unset.
+   *
+   * However, there are no values we can use as the default for that property
```

Comment:
And no default doesn't work either?

I know some very unintuitive things happen because `var()` is just string substitution and whitespace is a valid CSS value. I wonder if:

```css
      :host {
        background-color:var(--playground-editor-background-color);,
      }
```

(no whitespace) would work?

Also, what selector is code mirror using to set the background? It's presumably not using :host(), so there must be some other element for it to select, and I wonder why that background isn't sufficient if the host is `unset` or `transparent`?

Reply:
The whitespace idea doesn't work, it always gets evaluated to transparent when it's unset.

The selector is determined by the theme stylesheet, but it will be something like something like `.cm-s-monokai.CodeMirror`. So, to override that, we have to set the `background-color` of `.CodeMirror`. However, we still want the default to be the theme's background-color, and there is no default value that will have that effect, because they all reset it to the inherited or default value, rather than acting as though there were no value set at all (which is what I thought "revert" would do, but it doesn't).

> and I wonder why that background isn't sufficient if the host is `unset` or `transparent`?

Not sure I understand this. We can't use `unset` or `transparent` because that will always override the theme background-color?